### PR TITLE
[explorer/nodewatch]: add initial NEM and Symbol clients along with NodeInfo model

### PR DESCRIPTION
**Pull request by Jaguar0625**

### CODE REVIEW

File: explorer/nodewatch/puller/model/NodeInfo.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+		# pylint: disable=protected-access
+		return isinstance(other, NodeInfo) and (
+			self.network_id == other.network_id and (
+				self.network_generation_hash_seed == other.network_generation_hash_seed and (
+					self.main_public_key == other.main_public_key and (
+						self.node_public_key == other.node_public_key and (
+							self.endpoint == other.endpoint and (
+								self.name == other.name and (
+									self.version == other.version and (
+										self.roles == other.roles))))))))
```
Comment (gimre-xymcity) [inline explorer/nodewatch/puller/model/NodeInfo.py:51]: oh my what happened with tabs here?

File: explorer/nodewatch/puller/model/NodeInfo.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+	def __eq__(self, other):
+		# pylint: disable=protected-access
+		return isinstance(other, NodeInfo) and (
+			self.network_id == other.network_id and (
+				self.network_generation_hash_seed == other.network_generation_hash_seed and (
+					self.main_public_key == other.main_public_key and (
+						self.node_public_key == other.node_public_key and (
+							self.endpoint == other.endpoint and (
+								self.name == other.name and (
```
Comment (gimre-xymcity) [inline explorer/nodewatch/puller/model/NodeInfo.py:49]: do we need/want name to actually be equal? (esp if everything else matches)

File: explorer/nodewatch/puller/client/NemClient.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		endpoint_dict = node_dict['endpoint']
+		return NodeInfo(
+			node_dict['metaData']['networkId'],
+			None,
+			None,  # TODO need to fill this in for node_info
+			PublicKey(node_dict['identity']['public-key']),
+			f'{endpoint_dict["protocol"]}://{endpoint_dict["host"]}:{endpoint_dict["port"]}',
+			node_dict['identity']['name'],
+			node_dict['metaData']['version'],
+			NodeInfo.API_ROLE_FLAG)
```
Comment (gimre-xymcity) [inline explorer/nodewatch/puller/client/NemClient.py:41]: always an api, right?

File: explorer/nodewatch/puller/client/NemClient.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+class NemClient(BasicClient):
+	"""Async client for connecting to a NEM node."""
+
+	async def chain_height(self):
+		"""Gets chain height."""
+
+		chain_height = await self.get('chain/height', 'height')
+		return int(chain_height)
```
Comment (gimre-xymcity) [inline explorer/nodewatch/puller/client/NemClient.py:14]: I'm guessing this (and in symbolclient below) will be Height?

File: explorer/nodewatch/puller/client/SymbolClient.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+	async def peers(self):
+		"""Gets peers network."""
+
+		nodes_dict = await self.get('node/peers')
+		return [self._map_to_node_info(node_dict) for node_dict in nodes_dict]
+
+	@staticmethod
+	def _map_to_node_info(node_dict):
+		# TODO: need to filter nodes from wrong network somehow
```
Comment (gimre-xymcity) [inline explorer/nodewatch/puller/client/SymbolClient.py:55]: as mentioned, imo filtering by network probably should not happen at this level

File: explorer/nodewatch/puller/client/SymbolClient.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+		# TODO: need to detect HTTPS somehow
```
Comment (gimre-xymcity) [inline explorer/nodewatch/puller/client/SymbolClient.py:56]: ofc, there's no way currently to detect it by response only...
(also I don't think we'll have a way to do it, without making multiple requests)

File: explorer/nodewatch/tests/model/test_NodeInfo.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+	def test_repr_is_supported(self):
+		# Arrange:
+		info = self._create_default_info()
+
+		# Act:
+		info_repr = repr(info)
+		info2 = eval(info_repr)  # pylint: disable=eval-used
+
+		# Assert:
+		self.assertEqual('\n'.join([
```
Comment (gimre-xymcity) [inline explorer/nodewatch/tests/model/test_NodeInfo.py:119]: hmm, wouldn't it have more sense to just wrap expected thing in multi-line string `"""`?

File: explorer/nodewatch/puller/model/NodeInfo.py (hunk 1 above)
Comment (Jaguar0625) [reply explorer/nodewatch/puller/model/NodeInfo.py:51]: pylint - that was only way i could get it to stop complaining

File: explorer/nodewatch/puller/model/NodeInfo.py (hunk 2 above)
Comment (Jaguar0625) [reply explorer/nodewatch/puller/model/NodeInfo.py:49]: i don't mind, we can probably remove that

File: explorer/nodewatch/puller/client/NemClient.py (hunk 4 above)
Comment (Jaguar0625) [reply explorer/nodewatch/puller/client/NemClient.py:14]: we could wrap it in pod from models

File: explorer/nodewatch/tests/model/test_NodeInfo.py (hunk 7 above)
Comment (Jaguar0625) [reply explorer/nodewatch/tests/model/test_NodeInfo.py:119]: no real preference, but i generally find this a bit more readable, as the ''' string would need to be deindented relative to the code

**Review by gimre-xymcity — DISMISSED** [review]

File: explorer/nodewatch/puller/client/NemClient.py [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+	async def node_info(self):
+		"""Gets node information."""
+
+		node_dict = await self.get('node/info')
+		node_info = self._map_to_node_info(node_dict)
+
+		# lookup main account public key
+		network = NetworkLocator.find_by_identifier(Network.NETWORKS, node_info.network_identifier)
```
[the remark this answers is not in the document]
Comment (Jaguar0625) [reply explorer/nodewatch/puller/client/NemClient.py:45]: you mean via constructor?

**Review by gimre-xymcity — DISMISSED** [review]

**Pull request merged**